dashboard/app/models/level_source.rb

```diff
@@ -51,4 +51,26 @@ def self.find_identical_or_create(level, data)
       end
     end
   end
+
+  # @param [Integer] user_id The ID of the user performing the obfuscation.
+  # @return [String] The obfuscated (with the user_id) level source ID.
+  def obfuscate_level_source_id(user_id)
+    Base64.urlsafe_encode64("#{id}:#{user_id}")
+  end
+
+  # @param [String] obfuscated_level_source_id_user_id The obfuscated (with the user_id)
+  #   level_source ID.
+  # @param [Boolean] ignore_missing_user Whether to ignore the absence of the user specified by the
+  #   user ID when returning the level source ID. Default false.
+  # @return [Integer, nil] The (deobfuscated) level source ID, returns nil if the user specified in
+  #   the obfuscation does not exist.
+  def self.deobfuscate_level_source_id(obfuscated_level_source_id_user_id, ignore_missing_user: false)
+    level_source_id, user_id = Base64.urlsafe_decode64(obfuscated_level_source_id_user_id).split(':')
+    if ignore_missing_user || user_id.nil? || User.find_by_id(user_id)
```

Comment:
We need to support the use case of a non-signed in user sharing a level source. This is the workflow where we expect user_id to be nil. Note that we could instead share the /c/ link, though this seems better as it allows us to eventually deprecate the /c/ link if desired.

Yes. This is also true for a signed-in user, guessing the ID of a non-deleted signed-in user. In either case, the level source could be accessed by hitting the /c/ link (which still remains), so security by obfuscation is as best as we can get.

If we wanted, we could add a secret into the (de)obfuscation. Again, though, since the content is accessible through the /c/ link, this seems overkill.